python/ray/dag/compiled_dag_node.py

```diff
@@ -1908,7 +1908,7 @@ def teardown(self, kill_actors: bool = False):
                     _destroy_nccl_group(nccl_group_id)
 
                 logger.info("Waiting for worker tasks to exit")
-                self.wait_teardown()
+                self.wait_teardown(kill_actors=kill_actors)
```

Comment:
Do you really need to kill the actors? When the driver exits, won't it clean up ray resources?

Reply:
ray's shutdown calls 
https://github.com/ray-project/ray/blob/cdaeaaef672a49568eed515f097cc068958a648b/python/ray/dag/compiled_dag_node.py#L90-L95

We just never passed the kill_actors all the way through to wait_teardown before. It can hang otherwise here on shutdown if we have tasks that don't end